geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/CacheClientNotifier.java

```diff
@@ -1564,6 +1564,8 @@ private void closeDeadProxies(List deadProxies, boolean stoppedNormally) {
       } catch (CancelException e) {
         throw e;
       } catch (Exception e) {
+        logger.warn("CacheClientNotifier: Caught exception attempting to close client: {}", proxy,
```

Comment:
Hi Barry, I was going through the logger.warn signatures and I found
`/**

     * Logs a message at the {@link Level#WARN WARN} level including the stack trace of the {@link Throwable}
     * <code>t</code> passed as parameter.
     *
     * @param message the message object to log.
     * @param t the exception to log, including its stack trace.
     */
    void warn(String message, Throwable t);`
and 
`    /**

     * Logs a message with parameters at warn level.
     *
     * @param message the message to log; the format depends on the message factory.
     * @param p0 parameter to the message.
     * @param p1 parameter to the message.
     */
    void warn(String message, Object p0, Object p1);`
I think the stacktrace will not be printed in this case as there is no extra parameter in the string.

maybe `logger.warn("CacheClientNotifier: Caught exception attempting to close client: " + proxy , e);`

Reply:
Funny. I had the same thought, but when I tried it, it logged the exception stack.

Here is an example:
```
[vm1] [warn 2019/10/14 17:36:55.403 PDT <ServerConnection on port 60696 Thread 4> tid=0x51] CacheClientNotifier: Caught exception attempting to close client: CacheClientProxy[identity(192.168.1.4(15458:loner):60721:33abd8cc,connection=1,durableAttributes=DurableClientAttributes[id=testRemoteHostAddress_2; timeout=300]); port=60722; primary=true; version=GEODE 1.11.0]
[vm1] java.lang.IllegalStateException: test
[vm1] 	at org.apache.geode.internal.cache.tier.sockets.CacheClientNotifier.closeDeadProxies(CacheClientNotifier.java:1564)
[vm1] 	at org.apache.geode.internal.cache.tier.sockets.CacheClientNotifier.unregisterClient(CacheClientNotifier.java:567)
[vm1] 	at org.apache.geode.internal.cache.tier.sockets.ClientHealthMonitor.unregisterClient(ClientHealthMonitor.java:255)
[vm1] 	at org.apache.geode.internal.cache.tier.sockets.ServerConnection.handleTermination(ServerConnection.java:961)
[vm1] 	at org.apache.geode.internal.cache.tier.sockets.ServerConnection.handleTermination(ServerConnection.java:882)
[vm1] 	at org.apache.geode.internal.cache.tier.sockets.ServerConnection.run(ServerConnection.java:1223)
[vm1] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[vm1] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[vm1] 	at org.apache.geode.internal.cache.tier.sockets.AcceptorImpl.lambda$initializeServerConnectionThreadPool$3(AcceptorImpl.java:674)
[vm1] 	at org.apache.geode.internal.logging.LoggingThreadFactory.lambda$newThread$0(LoggingThreadFactory.java:119)
[vm1] 	at java.lang.Thread.run(Thread.java:748)
```
Thats because the Message being logged is a GemFireParameterizedMessage, and that handles the above case.

It takes the last parameter and stores it in the throwable field if the last parameter is a throwable and there are more parameters than the message is calling for.

Checkout GemFireParameterizedMessage.parseArguments for details.
```
[vm1] GemFireParameterizedMessageFactory.newMessage params=[CacheClientProxy[identity(192.168.1.4(15458:loner):60721:33abd8cc,connection=1,durableAttributes=DurableClientAttributes[id=testRemoteHostAddress_2; timeout=300]); port=60722; primary=true; version=GEODE 1.11.0], java.lang.IllegalStateException: test]
[vm1] GemFireParameterizedMessage.parseArguments argsCount=1; resultArgCount=2
[vm1] GemFireParameterizedMessage.parseArguments throwable = java.lang.IllegalStateException: test
[vm1] GemFireParameterizedMessage.GemFireParameterizedMessage stringArgs=[CacheClientProxy[identity(192.168.1.4(15458:loner):60721:33abd8cc,connection=1,durableAttributes=DurableClientAttributes[id=testRemoteHostAddress_2; timeout=300]); port=60722; primary=true; version=GEODE 1.11.0]]
```